server/grpc_service.go

```diff
@@ -300,9 +300,7 @@ func (s *GrpcServer) GetMinTS(
 		if done, err := limiter.Allow(fName); err == nil {
 			defer done()
 		} else {
-			return &pdpb.GetMinTSResponse{
-				Header: s.wrapErrorToHeader(pdpb.ErrorType_UNKNOWN, err.Error()),
-			}, nil
+			return nil, status.Error(codes.ResourceExhausted, err.Error())
```

Comment:
Maybe we can split these updates to another pr, and merge it firstly. It is simple